libdnf-plugins/actions/actions.cpp

```diff
@@ -0,0 +1,725 @@
+                if (act.args.empty()) {
+                    throw ActionsPluginError(
+                        M_("Error in file \"{}\" at line {}: Missing command"), path.native(), line_number);
+                }
+                act.command = act.args[0];
+
+                switch (hook) {
+                    case Hooks::PRE_BASE_SETUP:
+                        pre_base_setup_actions.emplace_back(std::move(act));
+                        break;
+                    case Hooks::POST_BASE_SETUP:
+                        post_base_setup_actions.emplace_back(std::move(act));
+                        break;
+                    case Hooks::PRE_TRANS:
+                        pre_trans_actions.emplace_back(std::move(act));
+                        break;
+                    case Hooks::POST_TRANS:
+                        post_trans_actions.emplace_back(std::move(act));
+                }
+            }
+        }
+    }
+}
+
+void Actions::process_command_output_line(std::string_view line) {
+    auto eq_pos = line.find('=');
+    if (line.starts_with("tmp.")) {
+        std::string var_name(line.substr(4, eq_pos - 4));
+        if (eq_pos == std::string::npos) {
+            tmp_variables.erase(var_name);
+        } else {
+            tmp_variables[var_name] = line.substr(eq_pos + 1);
+        }
+        return;
+    }
+    if (eq_pos == std::string::npos) {
+        base.get_logger()->error(
+            "Actions plugin: Syntax error: Command returns line without eq character \"{}\"", std::string(line));
+        return;
+    }
+    if (line.starts_with("conf.")) {
+        std::string var_name(line.substr(5, eq_pos - 5));
+        std::string var_value(line.substr(eq_pos + 1));
+        auto config_opts = base.get_config().opt_binds();
+        auto it = config_opts.find(var_name);
+        if (it == config_opts.end()) {
+            base.get_logger()->error("Actions plugin: Command returns unknown config option \"{}\"", var_name);
+            return;
+        }
+        try {
+            it->second.new_string(libdnf::Option::Priority::PLUGINCONFIG, var_value);
+        } catch (libdnf::OptionError & ex) {
+            base.get_logger()->error(
+                "Actions plugin: Cannot set config value returned by command \"{}={}\": {}",
+                var_name,
+                var_value,
+                std::string(ex.what()));
+        }
+    } else if (line.starts_with("var.")) {
+        std::string var_name(line.substr(4, eq_pos - 4));
+        std::string var_value(line.substr(eq_pos + 1));
+        base.get_vars()->set(var_name, var_value, libdnf::Vars::Priority::PLUGIN);
+    } else {
+        base.get_logger()->error("Actions plugin: Command returns unknown line \"{}\"", std::string(line));
+    }
+}
+
+void Actions::execute_command(CommandToRun & command) {
+    int pipe_out_from_child[2];
+    int pipe_to_child[2];
+    if (pipe(pipe_to_child) == -1) {
+        auto errnum = errno;
```

Comment:
This temporary variable is unnecessary, right? Repeated two more times below.

Reply:
OK